data-prepper-core/src/main/java/com/amazon/dataprepper/parser/model/DataPrepperConfiguration.java

```diff
@@ -0,0 +1,64 @@
+package com.amazon.dataprepper.parser.model;
+
+import java.io.File;
+import java.io.IOException;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+/**
+ * Class to hold configuration for DataPrepper, including server port and Log4j settings
+ */
+public class DataPrepperConfiguration {
+    private int serverPort = 4900;
+    private Log4JConfiguration log4JConfiguration = Log4JConfiguration.DEFAULT_CONFIG;
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper(new YAMLFactory());
+
+    public static final DataPrepperConfiguration DEFAULT_CONFIG = new DataPrepperConfiguration();
+
+    /**
+     * Construct a DataPrepperConfiguration from a yaml file
+     * @param file
+     * @return
+     */
+    public static DataPrepperConfiguration fromFile(File file) {
+        try {
+            return OBJECT_MAPPER.readValue(file, DataPrepperConfiguration.class);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Invalid DataPrepper configuratino file.");
+        }
+    }
+
+    private DataPrepperConfiguration() {}
+
+    @JsonCreator
+    public DataPrepperConfiguration(
+            @JsonProperty("serverPort") final int serverPort,
+            @JsonProperty("log4jConfig") final Log4JConfiguration log4JConfiguration
+    ) {
+        setServerPort(serverPort);
+        setLog4JConfiguration(log4JConfiguration);
+    }
+
+    private void setServerPort(int serverPort) {
+        if(serverPort != 0) {
+            this.serverPort = serverPort;
+        }
+    }
+
+    private void setLog4JConfiguration(Log4JConfiguration log4JConfiguration) {
+        if(log4JConfiguration != null) {
+            this.log4JConfiguration = log4JConfiguration;
+        }
+    }
```

Comment:
So the "!= null" case here is so that we don't override the default to null when nothing is passed in. The ObjectMapper YAML parser will pass in "null" to the constructor if nothing is in the YAML. 

However, I think you're right that we should have some validation on this configuration

Reply:
Ah gotcha, didn't realize these were being populated by a mapper (hence the null checks) 👍